### PR TITLE
simplify component tests

### PR DESCRIPTION
**Pull request by justinpjose**
### What
**Describe what you have changed and why.**
- Remove the step `the search api is working correctly` in all the other features except for the `posting_a_job.feature`
- This is because the Search API is only used in the `CreateJob` handler and not in any of the others
- This was achieved by using the mongo client within the component test rather than doing a `POST /jobs` request within the component test to generate jobs in the datastore (`mongo`). 
- This makes it more performant as it is now only doing the create job in `mongo` when asked whereas the component test would have previously done the `POST /jobs` which would do additional processes such as creating an index and getting the index name which was not needed
- This also simplifies and removes code that reads the HTTP response body from those calls which is no longer required 

### How to review
**Describe the steps required to test the changes.**
- Check if the code changes make sense
- Check if the tests pass

### Who can review
**Describe who worked on the changes, so that other people can review.**
Anyone

### CODE REVIEW

**Review by cookel2 — COMMENTED** [review]
Just one suggestion but see what you think.

File: features/latest_version_features/getting_a_job.feature [hunk 1]
```diff
@@ -2,8 +2,7 @@ Feature: Getting a job
 
   Scenario: Job exists in the Job Store and a get request returns it successfully
 
-    Given the search api is working correctly
-    And set the api version to undefined for incoming requests
+    Given set the api version to undefined for incoming requests
```
[the remark this answers is not in the document]
Comment (justinpjose) [reply features/latest_version_features/getting_a_job.feature:5]: That's a good suggestion. I thought of doing this in multiple places (not only for this step but for others as well). I will do this in this PR then

File: features/latest_version_features/posting_a_job.feature [hunk 2]
```diff
@@ -3,11 +3,12 @@ Feature: Posting a job
   Scenario: Job is posted successfully
 
     Given the search api is working correctly
-    And set the api version to undefined for incoming requests
+    Given the api version is undefined for incoming requests
```
Comment (gilesburdett) [inline features/latest_version_features/posting_a_job.feature:6]: How does the `Given` on line 6 interact with the `Given` on line 5?

File: features/v1_features/latest_version_features/posting_a_job.feature [hunk 3]
```diff
@@ -3,32 +3,32 @@ Feature: Posting a job
   Scenario: Job is posted successfully
 
     Given the search api is working correctly
-    And set the api version to v1 for incoming requests
+    Given the api version is v1 for incoming requests
```
Comment (gilesburdett) [inline features/v1_features/latest_version_features/posting_a_job.feature:6]: A second instance of the double `Given`

File: features/latest_version_features/posting_a_job.feature (hunk 2 above)
Comment (justinpjose) [reply features/latest_version_features/posting_a_job.feature]: Good spot. Addressed

File: features/v1_features/latest_version_features/posting_a_job.feature (hunk 3 above)
Comment (justinpjose) [reply features/v1_features/latest_version_features/posting_a_job.feature]: Good spot. Addressed

File: features/latest_version_features/getting_a_job.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (justinpjose) [reply features/latest_version_features/getting_a_job.feature]: Done

**Review by cookel2 — APPROVED** [review]
looks good to me

**Pull request merged**